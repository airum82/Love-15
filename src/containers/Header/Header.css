.header button {
  background-color: #60b0f4;
  color: white; 
  border: none;
  height: 6vh;
  text-decoration: none;
  width: 10vw; 
}

.log-in-button {
  display: grid;
  grid-template-columns: 40% 40%;
  width: 30%;
  float: right;
  margin: 1% 0;
}

header {
  background-color: #003fff;
  height: 9vh;
  position: fixed;
  width: 100%;
}

.welcome p {
  display: inline;
  margin-right: 1%;
  font-size: 1.3em;
}

.welcome {
  margin-top: 1%;
}

.welcome {
  text-align: left;
}

.header img {
  height: 30vh;
  margin-top: 8%;
}

h3 {
  margin: 0 auto;
  font-size: 3.5vh;
  color: #60b0f4;
  width: 25%;
}