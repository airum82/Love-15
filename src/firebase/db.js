import { db } from './firebase';

export const doCreateUser = (id, username, email) =>
  db.ref(`users/${id}`).set({
    username,
    email,
  });

export const onceGetUsers = () =>
  db.ref('users').once('value');

export const addFavoriteCourt = (id, court) => {
  return db.ref(`users/${id}/favorites`).push(court)
}

export const grabFavoriteCourtsList = (id) => {
  return db.ref(`users/${id}/favorites`).once('value')
}

export const removeFavoriteFromList = (userId, courtId) => {
  db.ref(`users/${userId}/favorites/${courtId}`).remove()
}