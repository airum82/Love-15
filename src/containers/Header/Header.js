import React from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import logo from '../../images/Love-15-logo.png';
import { logOut, clearUserList, clearFavorites } from '../../actions';
import './Header.css';
import { connect } from 'react-redux';
import { auth } from '../../firebase';
import PropTypes from 'prop-types';


export const Header = (props) => {
  return (
    <div className="header">
      <header>
        {props.account.email ? '' :
          <div className="log-in-button">
            <NavLink to='/logIn'>
              <button>Log In</button>
            </NavLink>
            <NavLink to='/createAccount' className="create">
              <button>Create Account</button>
            </NavLink>
          </div>
        }
        {props.account.email ? 
          <div className= "welcome">
            <p>Welcome {props.account.email}</p>
            <button className="log-out" onClick={() => {
              auth.doSignOut()
                .then(signOut => props.handleLogOut())
                .then(signOut => props.clearUserList())
                .then(signOut => props.clearFavorites());
              props.history.push('/');
            }}>Log Out</button>
            <NavLink to='/UserList'>
              <button>View Users</button>
            </NavLink>
            <NavLink to='/favorites'>
              <button>View Favorites</button>
            </NavLink>
          </div> 
          : ''}
      </header>
      <NavLink to='/' style={{ textDecoration: 'none' }}>
        <img src={logo} className="logo"/>
        <h3>It's time to rally!</h3>
      </NavLink>
    </div>
  );
};

export const mapStateToProps = (state) => ({
  account: state.account
});

export const mapDispatchToProps = (dispatch) => ({
  handleLogOut: () => dispatch(logOut()),
  clearUserList: () => dispatch(clearUserList()),
  clearFavorites: () => dispatch(clearFavorites())
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));

Header.propTypes = {
  account: PropTypes.object,
  handleLogOut: PropTypes.func,
  clearUserList: PropTypes.func,
  clearFavorites: PropTypes.func,
  history: PropTypes.object
};