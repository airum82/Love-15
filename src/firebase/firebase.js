import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore';
import { firebaseKey } from '../APIkey';
import 'firebase/database';

const config = {
  apiKey: firebaseKey,
  authDomain: "love-15-210720.firebaseapp.com",
  databaseURL: "https://love-15-210720.firebaseio.com",
  projectId: "love-15-210720",
  storageBucket: "love-15-210720.appspot.com",
  messagingSenderId: "57178198843"
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const db = firebase.database();
const auth = firebase.auth();

export {
  db,
  auth,
};