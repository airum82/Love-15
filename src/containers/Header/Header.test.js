import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { auth } from '../../firebase';

describe('Header', () => {
  it('should match snapshot upon render', () => {
    const wrapper = shallow(
      <Header
       account={{ email: 'billy@billy.net' }} 
      />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should call auth.doSignOut on click', () => {
    auth.doSignOut = jest.fn().mockImplementation(() => Promise.resolve());
    const wrapper = shallow(
      <Header
        account={{ email: 'billy@billy.net' }}
        handleLogOut={jest.fn()}
        clearUserList={jest.fn()}
        clearFavorites={jest.fn()}
        history={{
          push: jest.fn()
        }}
      />);

    wrapper.find('.welcome').find('.log-out').simulate('click');
    expect(auth.doSignOut).toHaveBeenCalled();
  })

  it('mapStateToProps should return an object with appropriate keys', () => {
    const mockState = {
      account: {
        email: 'billybob@truckhat.org'
      }
    }
    const result = mapStateToProps(mockState);
    const expectedOutput = {
      account: {
        email: 'billybob@truckhat.org'
      }
    }
    expect(result).toEqual(expectedOutput);
  })

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let mappedProps;
    beforeEach(() => {
      mockDispatch = jest.fn();
      mappedProps = mapDispatchToProps(mockDispatch);
    })

    it('handleLogOut should call dispatch', () => {
      mappedProps.handleLogOut();
      expect(mockDispatch).toHaveBeenCalled();
    })

    it('clearUserList should call dispatch', () => {
      mappedProps.clearUserList();
      expect(mockDispatch).toHaveBeenCalled();
    })

    it('clearFavorites should call dispatch', () => {
      mappedProps.clearFavorites();
      expect(mockDispatch).toHaveBeenCalled();
    })
  })
})