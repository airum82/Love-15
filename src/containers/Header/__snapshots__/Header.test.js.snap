// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot upon render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          style={
                    Object {
                              "textDecoration": "none",
                            }
          }
          to="/"
>
          <img
                    className="logo"
                    src="love-15-logo2.png"
          />
          <h3>
                    It's time to rally!
          </h3>
</NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/logIn"
>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
</NavLink>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": Array [
            <img
              className="logo"
              src="love-15-logo2.png"
/>,
            <h3>
              It's time to rally!
</h3>,
          ],
          "style": Object {
            "textDecoration": "none",
          },
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo",
              "src": "love-15-logo2.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "It's time to rally!",
            },
            "ref": null,
            "rendered": "It's time to rally!",
            "type": "h3",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          "to": "/logIn",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            style={
                        Object {
                                    "textDecoration": "none",
                                  }
            }
            to="/"
>
            <img
                        className="logo"
                        src="love-15-logo2.png"
            />
            <h3>
                        It's time to rally!
            </h3>
</NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/logIn"
>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
</NavLink>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": Array [
              <img
                className="logo"
                src="love-15-logo2.png"
/>,
              <h3>
                It's time to rally!
</h3>,
            ],
            "style": Object {
              "textDecoration": "none",
            },
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "logo",
                "src": "love-15-logo2.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "It's time to rally!",
              },
              "ref": null,
              "rendered": "It's time to rally!",
              "type": "h3",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            "to": "/logIn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
